packages/amesos2/src/Amesos2_TpetraCrsMatrix_MatrixAdapter_def.hpp

```diff
@@ -366,9 +381,13 @@ namespace Amesos2 {
                 int row = perm_g2l(i);
                 for (int k=pointers_t(i); k<pointers_t(i+1); k++) {
                   int col = indices_t(k);
-                  transpose_map(k) = pointers(1+row);
-                  indices(pointers(1+row)) = col;
-                  pointers(1+row) ++;
+                  if (col < nRows) {
+                    transpose_map(k) = pointers(1+row);
+                    indices(pointers(1+row)) = col;
+                    pointers(1+row) ++;
+                  } else {
+                    transpose_map(k) = -1;
```

Comment:
It looks like `transpose_map` is a templated type `host_ordinal_type_array`, is this always a signed type (since assigning -1 to it)?

Reply:
@ndellingwood. `host_ordinal_type_array` is defined using `local_ordinal_type`, e.g.,

https://github.com/trilinos/Trilinos/blob/8448d1ad45d37dc4cfddfd4998d944bf66cb1318/packages/amesos2/src/Amesos2_KLU2_decl.hpp#L204

So, this should be okay?

Reply:
Yeah, I think `int` is the supported `local_ordinal_type`, thanks for clarifying